From d20473c8631f9cda8c0a8563fbdb96d4287821cf Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Tue, 24 Oct 2017 02:27:29 +0200
Subject: [PATCH 08/12] msm8974-common: libril: Add workaround for "ring of
 death" bug

[javelinanddart]: Forward port to ril-caf on lineage-15.1

Change-Id: Ifeeee11ee83deddb8eff565847641c45a105378f
---
 ril/libril/ril_service.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/ril/libril/ril_service.cpp b/ril/libril/ril_service.cpp
index c38ae25..1e21397 100644
--- a/ril/libril/ril_service.cpp
+++ b/ril/libril/ril_service.cpp
@@ -7727,6 +7727,18 @@ int radio::cdmaInfoRecInd(int slotId,
                     record->signal[0].signalType = infoRec->rec.signal.signalType;
                     record->signal[0].alertPitch = infoRec->rec.signal.alertPitch;
                     record->signal[0].signal = infoRec->rec.signal.signal;
+
+                    /* Drop the response to workaround the "ring of death" bug */
+                    if (infoRec->rec.signal.isPresent
+                            /* IS95_CONST_IR_SIGNAL_IS54B */
+                            && infoRec->rec.signal.signalType == 2
+                            /* IS95_CONST_IR_ALERT_MED */
+                            && infoRec->rec.signal.alertPitch == 0
+                            /* IS95_CONST_IR_SIG_IS54B_L */
+                            && infoRec->rec.signal.signal == 1) {
+                        return 0;
+                    }
+
                     break;
                 }
 
-- 
2.7.4

